metrics/segmentStats.py

```diff
@@ -0,0 +1,360 @@
+        # improper binarization, so remove them and do not account for statistics
+        self.typeGraphdict[self._ithDisjointGraph] = 2
+        listOfPerms = list(itertools.combinations(nodes, 2))
+        if type(nodes[0]) == int:
+            modulus = [[start - end] for start, end in listOfPerms]
+            dists = [abs(i[0]) for i in modulus]
+        else:
+            dims = len(nodes[0])
+            modulus = [[start[dim] - end[dim] for dim in range(0, dims)] for start, end in listOfPerms]
+            dists = [sum(modulus[i][dim] * modulus[i][dim] for dim in range(0, dims)) for i in range(0, len(modulus))]
+        if len(list(nx.articulation_points(self._subGraphSkeleton))) == 1 and set(dists) != 1:
+            # each node is connected to one or two other nodes which are not a distance of 1 implies there is a
+            # one branch point with two end points in a single dichotomous tree"""
+            for sourceOnTree, item in listOfPerms:
+                if nx.has_path(self._subGraphSkeleton, sourceOnTree, item) and sourceOnTree != item:
+                    simplePaths = list(nx.all_simple_paths(self._subGraphSkeleton, source=sourceOnTree, target=item))
+                    simplePath = simplePaths[0]
+                    countBranchNodesOnPath = sum([1 for point in simplePath if point in nodes])
+                    if countBranchNodesOnPath == 2:
+                        curveLength = self._getLengthAndRemoveTracedPath(simplePath)
+                        self.isolatedEdgeInfoDict[sourceOnTree, item] = curveLength
+        else:
+            # each node is connected to one or two other nodes implies it is a line,
+            endPoints = [k for (k, v) in self._nodeDegreeDict.items() if v == 1]
+            sourceOnLine = endPoints[0]
+            targetOnLine = endPoints[1]
+            simplePath = nx.shortest_path(self._subGraphSkeleton, source=sourceOnLine, target=targetOnLine)
+            curveLength = self._getLengthAndRemoveTracedPath(simplePath)
+            self.isolatedEdgeInfoDict[sourceOnLine, targetOnLine] = curveLength
+
+    def _setCountDict(self, source):
+        if source not in self._visitedSources:
+            self.countDict[source] = 1
+            self._visitedSources.append(source)
+        else:
+            self.countDict[source] = self.countDict[source] + 1
+
+    def _setHausdorffDimensionDict(self, source, target, length, displacement):
+        logDisplacement = np.log(displacement)
+        if logDisplacement != -np.inf or not np.allclose(logDisplacement, 0.0):
+            self.hausdorffDimensionDict[self.countDict[source], source, target] = np.log(length) / logDisplacement
+
+    def _singleCycle(self, cycle):
+        """
+        Find statistics of a single cycle of a disjoint graph
+        """
+        sourceOnCycle = cycle[0]
+        self._setCountDict(sourceOnCycle)
+        curveLength = self._getLengthAndRemoveTracedPath(cycle, isCycle=1)
```

Comment:
`isCycle=1` --> `isCycle=True`

This isn't C! :-P  It does look like you've cleaned up most of those already, though.